src/core/service.c

```diff
@@ -4157,6 +4202,20 @@ static int service_dispatch_timer(sd_event_source *source, usec_t usec, void *us
                 service_enter_running(s, SERVICE_SUCCESS);
                 break;
 
+        case SERVICE_MOUNTING:
+                log_unit_warning(UNIT(s), "Mount operation timed out. Killing mount process.");
+                service_kill_control_process(s);
+                s->mount_result = SERVICE_FAILURE_TIMEOUT;
+                if (s->mount_request) {
+                        (void) sd_bus_reply_method_errorf(s->mount_request, SD_BUS_ERROR_TIMEOUT,
+                                                          "Method '%s' for unit '%s' timed out",
+                                                          strna(sd_bus_message_get_member(s->mount_request)),
+                                                          UNIT(s)->id);
+                        s->mount_request = sd_bus_message_unref(s->mount_request);
+                }
```

Comment:
please add a helper that replies to the method call either positively or negatively, and then call it here, and above.